modules/kargoBidAdapter.js

```diff
@@ -4,286 +4,484 @@ import { registerBidder } from '../src/adapters/bidderFactory.js';
-      device: {
-        width: window.screen.width,
-        height: window.screen.height
-      },
-      prebidRawBidRequests: validBidRequests
-    }, spec._getAllMetadata(bidderRequest, tdid));
-
-    // User Agent Client Hints / SUA
-    const uaClientHints = deepAccess(firstBidRequest, 'ortb2.device.sua');
-    if (uaClientHints) {
-      transformedParams.device.sua = pick(uaClientHints, ['browsers', 'platform', 'mobile', 'model']);
-    }
+  if (currency != null && currency != CURRENCY.US_DOLLAR) {
+    krakenParams.cur = currency
+  }
 
-    // Pull Social Canvas segments and embed URL
-    const socialCanvas = deepAccess(firstBidRequest, 'params.socialCanvas');
-    if (socialCanvas) {
-      transformedParams.socialCanvasSegments = socialCanvas.segments;
-      transformedParams.socialEmbedURL = socialCanvas.embedURL;
-    }
+  // Pull Social Canvas segments and embed URL
+  const socialCanvas = deepAccess(firstBidRequest, REQUEST_KEYS.SOCIAL_CANVAS);
 
-    const encodedParams = encodeURIComponent(JSON.stringify(transformedParams));
-    return Object.assign({}, bidderRequest, {
-      method: 'GET',
-      url: `${HOST}/api/v2/bid`,
-      data: `json=${encodedParams}`,
-      currency: currency
-    });
-  },
-  interpretResponse: function(response, bidRequest) {
-    let bids = response.body;
-    const bidResponses = [];
-    for (let bidId in bids) {
-      let adUnit = bids[bidId];
-      let meta = {
-        mediaType: BANNER
-      };
-
-      if (adUnit.metadata && adUnit.metadata.landingPageDomain) {
-        meta.clickUrl = adUnit.metadata.landingPageDomain[0];
-        meta.advertiserDomains = adUnit.metadata.landingPageDomain;
-      }
+  if (socialCanvas != null) {
+    krakenParams.socan = socialCanvas
+  }
+
+  // User Agent Client Hints / SUA
+  const uaClientHints = deepAccess(firstBidRequest, REQUEST_KEYS.SUA);
+  if (uaClientHints) {
+    const suaValidAttributes = []
 
-      if (adUnit.mediaType && SUPPORTED_MEDIA_TYPES.includes(adUnit.mediaType)) {
-        meta.mediaType = adUnit.mediaType;
+    for (const suaKey of SUA_ATTRIBUTES) {
+      const suaValue = uaClientHints[suaKey]
+      if(!suaValue) {
+        continue
       }
 
-      const bidResponse = {
-        ad: adUnit.adm,
-        requestId: bidId,
-        cpm: Number(adUnit.cpm),
-        width: adUnit.width,
-        height: adUnit.height,
-        ttl: 300,
-        creativeId: adUnit.id,
-        dealId: adUnit.targetingCustom,
-        netRevenue: true,
-        currency: adUnit.currency || bidRequest.currency,
-        mediaType: meta.mediaType,
-        meta: meta
-      };
-
-      if (meta.mediaType == VIDEO) {
-        bidResponse.vastXml = adUnit.adm;
+      // Do not pass any empty strings
+      if(typeof suaValue == 'string' && suaValue.trim() === '') {
+        continue
       }
 
-      bidResponses.push(bidResponse);
+      switch (suaKey) {
+        case SUA.MOBILE, SUA.SOURCE:
+          if (suaValue < 1) {
+            return;
```

Comment:
should this be `continue`?

Reply:
I initially thought the same but it turns out that continue will not work. In JavaScript forEach is a function so to end an iteration and move to the next you need to return.